rpcserver.go

```diff
@@ -1460,7 +1460,31 @@ func (r *rpcServer) SendPayment(paymentStream lnrpc.Lightning_SendPaymentServer)
 					// TODO(roasbeef): eliminate necessary
 					// encode/decode
 					nextPayment.Dest = payReq.Destination.SerializeCompressed()
-					nextPayment.Amt = int64(payReq.Amount)
+					if payReq.MilliSat == nil {
+						err := fmt.Errorf("only payment" +
+							" requests specifying" +
+							" the amount are" +
+							" currently supported")
+						select {
+						case errChan <- err:
+						case <-reqQuit:
+							return
+						}
+						return
+					}
+					nextPayment.Amt = int64(payReq.MilliSat.ToSatoshis())
+
+					if payReq.PaymentHash == nil {
```

Comment:
Shouldn't this logic be already pulled into the `Decode` function of `zpay32`?

Reply:
Correct, checked in the `validateInvoice` method within `Decode`. Removed check here and one other place in the file.